### PR TITLE
Rename hello world example to weather lookup

### LINKED ISSUE #1897: Weather service with Private Information Retrieval
As a new example of PIR in Oak - we can create an Oak application for anonymous weather information retrieval.
This application will accept user requests (that contain their location) and request weather information from an external service. As a simple example - for each request it can either download the whole database or split it in chunks.

The difference from existing Trusted Database example is that this application will download the database dynamically from an external service, whereas Trusted Database already keeps the static database in memory.

For requesting weather information we can use the following service: https://openweathermap.org/bulk

### PR DESCRIPTION
**Pull request by tiziano88**
Manually create tokio runtime and allow configuring the number of
threads in the config file. I noticed that when running on Cloud Run,
even fetching the lookup data file takes a long time (20s), and I
thought it may be because the runtime is actually running
single-threaded; by specifying the number of threads now the download
and parsing time seems to be down to <1s, which agrees with my theory

Fix #1897

# Checklist

- [ ] Pull request affects core Oak functionality (e.g. runtime, SDK, ABI)
  - [ ] I have written tests that cover the code changes.
  - [ ] I have checked that these tests are run by
        [Cloudbuild](/cloudbuild.yaml)
  - [ ] I have updated [documentation](/docs/) accordingly.
  - [ ] I have raised an [issue](https://github.com/project-oak/oak/issues) to
        cover any TODOs and/or unfinished work.
- [ ] Pull request includes prototype/experimental work that is under
      construction.

### CODE REVIEW

**Review by rbehjati — APPROVED** [review]
LGTM. Thanks!

File: oak_functions/examples/weather_lookup/README.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+       --manifest-path=./oak_functions/examples/weather_lookup/module/Cargo.toml \
+       --out-dir=./oak_functions/examples/weather_lookup/bin
+   ```
+
+1. Build `oak_functions_loader` (with debugging enabled if needed):
+
+   ```shell
+   cargo build --manifest-path=./oak_functions/loader/Cargo.toml \
+       --release \
+       --features=oak-unsafe
```
Comment (ipetr0v) [inline oak_functions/examples/weather_lookup/README.md:26]: Why do we need `unsafe`?
Is it to work with Wasm modules?

Comment (tiziano88) [reply oak_functions/examples/weather_lookup/README.md:26]: Just to have some logging I guess, it was already there.

Comment (rbehjati) [reply oak_functions/examples/weather_lookup/README.md:26]: Yes. It is for logging. We have to removed it for the prod version, I think.

**Pull request merged**